app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -99,4 +121,53 @@ public Retrofit getRetrofit() {
 
         return retrofit;
     }
+
+    /**
+     * Provides {@SecurePreferences} singleton instance to access shared preferences
+     *
+     * @return A single instance of {@SecurePreferences}
+     */
+    public static SharedPreferences getSharedPreferences() {
+        if (sSecurePreferences == null) {
+            synchronized (BaseApplication.class) {
+                if (sSecurePreferences == null) {
```

Comment:
The same `null` check has already been performed 2 lines earlier?

Reply:
That makes sure sSecurePreferences is not created mulitiple times in case there are many threads accessing the preferences at the same time. This is the basic implementation of Singleton.